modules/efk-logging-fluentd-undefined.adoc

```diff
@@ -0,0 +1,121 @@
+// Module included in the following assemblies:
+//
+// * logging/efk-logging-fluentd.adoc
+
+[id="efk-logging-fluentd-undefined_{context}"]
+= Configuring how the log collector normalizes logs 
+
+Cluster Logging uses the link:https://github.com/ViaQ/fluent-plugin-viaq_data_model[ViaQ common data model] to deal with problem that arise from conflicts 
+and inconsistencies in log data collected from different subsystems.   
+
+You can configure how defined, undefined, and empty log fields from disparate sources are treated by editing the log collector daemonset, Fluentd or Rsyslog, and setting environment variables in the table below. 
+
+// from https://github.com/ViaQ/fluent-plugin-viaq_data_model/commit/8b5ef11cedec4c372b2cb082afc7f9cc08473654
+
+* Undefined fields. One of the problems with log data from disparate systems is that some fields might be unknown to the ViaQ data model.
+Such fields are called _undefined_. ViaQ requires all top-level fields to be defined and described. 
++
+Use the parameters to configure how {product-title} moves any undefined fields under a top-level field called `undefined` to avoid conflicting with the _well known_ ViaQ 
+top-level fields. You can add undefined fields to the top-level fields and move others to an `undefined` container.
++
+You can also convert undefined fields to JSON format and replace special characters.
+
+* Defined fields. You can also configure which defined fields appear in the top levels of the logs. 
++
+[[default-fields]]
+The default top-level fields are `CEE,time`, `@timestamp`, `aushape`, `ci_job`, `collectd`, `docker`, `fedora-ci`, 
+`file`, `foreman`, `geoip`, `hostname`, `ipaddr4`, `ipaddr6`, `kubernetes`, `level`, `message`, `namespace_name`, `namespace_uuid`,
+`offset`, `openstack`, `ovirt`, `pid`, `pipeline_metadata`, `rsyslog`, `service`, `systemd`, `tags`, `testcase`, `tlog`, `viaq_msg_id`.
+You can use the `CDM_DEFAULT_KEEP_FIELDS` parameter to add additional defaults and the `CDM_EXTRA_KEEP_FIELDS` parameter to include 
+additional non-default fields at the top level.
++
+Any defined fields not included in `${CDM_DEFAULT_KEEP_FIELDS}` or `${CDM_EXTRA_KEEP_FIELDS}` are moved to `${CDM_UNDEFINED_NAME}` if `CDM_USE_UNDEFINED` is `true`.
```

Comment:
Suggestion: remove `defined` in `Any defined fields not included in`

Reply:
Outdated